src/test/java/bisq/core/dao/node/validation/GenesisTxValidatorTest.java

```diff
@@ -0,0 +1,100 @@
+import java.util.Optional;
+import org.bitcoinj.core.Coin;
+import org.junit.Test;
+import bisq.core.dao.state.blockchain.Tx;
+import bisq.core.dao.state.blockchain.RawTx;
+import bisq.core.dao.state.blockchain.RawTxOutput;
+import bisq.core.dao.state.blockchain.TxInput;
+import bisq.core.dao.state.blockchain.TxOutputType;
+import bisq.core.dao.state.blockchain.TxType;
+import com.google.common.collect.ImmutableList;
+
+public class GenesisTxValidatorTest {
+
+    @Test
+    public void testGetGenesisTx() {
+        int blockHeight = 200;
+        String blockHash = "abc123";
+        Coin genesisTotalSupply = Coin.parseCoin("2.5");
+        long time = new Date().getTime();
+        final List<TxInput> inputs = asList(
+                new TxInput("tx0", 0, null),
+                new TxInput("tx1", 1, null)
+        );
+        final List<RawTxOutput> outputs = asList(new RawTxOutput(0, 101, null, null, null, null, blockHeight));
+        RawTx rawTx = new RawTx(
+            "tx2",
+            blockHeight,
+            blockHash,
+            time,
+            ImmutableList.copyOf(inputs),
+            ImmutableList.copyOf(outputs)
+        );
+
+        String genesisTxId = "genesisTxId";
+        int genesisBlockHeight = 150;
+
+        // With mismatch in blockheight and tx id, we should not get genesis tx back.
+        Optional<Tx> result = GenesisTxValidator.getGenesisTx(genesisTxId, genesisBlockHeight, genesisTotalSupply, rawTx);
+        Optional<Tx> want = Optional.empty();
+        assertEquals(want, result);
+
+        // With correct blockheight but mismatch in tx id, we should still not get genesis tx back.
+        blockHeight = 150;
+        rawTx = new RawTx(
+            "tx2",
+            blockHeight,
+            blockHash,
+            time,
+            ImmutableList.copyOf(inputs),
+            ImmutableList.copyOf(outputs)
+        );
+        result = GenesisTxValidator.getGenesisTx(genesisTxId, genesisBlockHeight, genesisTotalSupply, rawTx);
+        want = Optional.empty();
+        assertEquals(want, result);
+
+        // With correct tx id and blockheight, we should find our genesis tx with correct tx and output type.
+        rawTx = new RawTx(
+            genesisTxId,
+            blockHeight,
+            blockHash,
+            time,
+            ImmutableList.copyOf(inputs),
+            ImmutableList.copyOf(outputs)
+        );
+        result = GenesisTxValidator.getGenesisTx(genesisTxId, genesisBlockHeight, genesisTotalSupply, rawTx);
+        Tx tx = new Tx(rawTx);
+        tx.setTxType(TxType.GENESIS);
+        for (int i = 0; i < tx.getTxOutputs().size(); ++i) {
+            tx.getTxOutputs().get(i).setTxOutputType(TxOutputType.GENESIS_OUTPUT);
+        }
+        want = Optional.of(tx);
+        assertEquals(want, result);
+        // TODO(chirhonul): test that only outputs in tx summing exactly to genesisTotalSupply is accepted, and
+        // that code under test raises RuntimeError otherwise.
```

Comment:
Maybe we should add a custom Exception there (extending RuntimeError)? Would make test more clear as well if u can check exception type.